plugin/src/main/java/io/jenkins/plugins/analysis/core/model/JobAction.java

```diff
@@ -202,6 +206,20 @@ public boolean isTrendVisible() {
         return trendChartType != TrendChartType.NONE && createBuildHistory().hasMultipleResults();
     }
 
+    public boolean noWarnings(){
+        History results =createBuildHistory(); //Get past build history
+        if(results instanceof AnalysisHistory){
+            Iterator<BuildResult<AnalysisBuildResult>> it = results.iterator();
+            while(it.hasNext()) {
+                BuildResult<AnalysisBuildResult> nextResult = it.next();
+                if(nextResult.getResult().getTotalSize() - nextResult.getResult().getTotalSizeOf(Severity.ERROR) != 0){
```

Comment:
All warnings are included in total. Or what exactly do you want to achieve with the subtraction?
```suggestion
                if (nextResult.getResult().getTotalSize() > 0) {
```

Reply:
Doesn't nextResult.getResult() returns the total number of warnings+errors? If we are only checking the number of warnings, shouldn't we subtract the number of errors from the total number?

Reply:
In case there is one error but no warnings, this new method is expected to return true, right?

Reply:
I noticed that you changed the method name to hasNoIssues(). Are we including errors then? I thought the graph only displayed the number of warnings😅.

Reply:
No, the chart draws "everything": actually the individual tools decide which things should be shown (I called them issues since these items can be warnings, errors, code duplications, TODO tags, unit test failures, etc.). So if at least one build has number of issues > 0, then the chart should be shown.

Reply:
Thank you for the clarification!